test/async-observer.test.js

```diff
@@ -324,6 +324,86 @@ describe('async observer', function() {
       }
     );
   });
+
+  it('should call after operation hook on error', function(done) {
+    const context = {
+      req: {},
+    };
+    const operationError = new Error('The operation failed without result');
+    let passesBy = 0;
```

Comment:
Let's use a more accurate name, e.g.

```suggestion
    let callCount = 0;
```

The same comment applies to other newly added tests below too.